jekyll/_cci2/executor-intro.adoc

```diff
@@ -0,0 +1,346 @@
+
+=== Images, containers, and virtual machines
+
+An _image_ is a packaged system that has the instructions for creating a running environment.
+
+A _container_ or _virtual machine_ is the term used for a running instance of an image.
+
+For example:
+
+* Jobs that require Docker images (`docker`) may use an image for Node.js, Python etc. The xref:circleci-images#[pre-built CircleCI Docker image] (convenience images) will help you get started. CircleCI convenience images are not a full operating system, so they generally make building your software more efficient.
+
+* Jobs that require a complete Linux virtual machine (VM) image (`machine`) may use an Ubuntu version supported by the xref:configuration-reference#available-linux-machine-images[list of available machine images].
+
+* Jobs that require a macOS VM image (`macos`) may use an Xcode version. A full list is available on the xref:using-macos#supported-xcode-versions[Using macOS] page.
+
+=== Resource class
+
+When you configure an executor it is recommended that you also specify a resource class to specify compute resource size (CPU and RAM).
+
+* For an overview of resource classes, see the xref:resource-class-overview#[Resource class overview] page.
+* For an overview of available resource classes per Plan, see the link:https://circleci.com/pricing/[Pricing page].
+
+[#docker]
+== Docker
+
+CAUTION: *Legacy images with the prefix "circleci/" were https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034[deprecated]* on December 31, 2021. For faster builds, upgrade your projects with link:https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/[next-generation convenience images].
+
+To access the Docker execution environment, use the `docker` executor and specify an image. For a full list of convenience images, which are built by CircleCI, see the link:https://circleci.com/developer/images?imageType=docker[CircleCI Developer Hub]
+
+[,yml]
+----
+jobs:
+  build: # name of your job
+    docker: # executor type
+      - image: cimg/base:stable # primary container will run the latest, production-ready base image
+
+    steps:
+        # Commands run in the primary container
+----
+
+Find out more about the Docker execution environment on the link:/docs/using-docker/[Using Docker] page.
+
+[#linux-vm]
+== Linux VM
+
+NOTE: *CircleCI cloud* The use of `machine: true` is deprecated. You must specify an image to use.
+
+To access the Linux VM execution environment, use the `machine` executor and specify a Linux image. For a full list of `machine` images, see the link:https://circleci.com/developer/images?imageType=machine[CircleCI Developer Hub]
+
+[.tab.machine.Cloud]
+--
+[,yml]
+----
+jobs:
+  build: # name of your job
+    machine: # executor type
+      image: ubuntu-2004:202010-01 # # recommended linux image - includes Ubuntu 20.04, docker 19.03.13, docker-compose 1.27.4
```

Comment:
Would it be better here to use `ubuntu-2004:current` or `ubuntu-2004:default`? 
We could also do `ubuntu-2204:current` or `ubuntu-2204:default`